compiler/code-gen/declarations.cpp

```diff
@@ -687,6 +691,36 @@ void ClassDeclaration::compile_back_includes(CodeGenerator &W, IncludesCollector
   W << includes;
 }
 
+void ClassDeclaration::compile_job_worker_shared_memory_piece_methods(CodeGenerator &W, bool compile_declaration_only) const {
+  auto request_interface = G->get_class("KphpJobWorkerRequest");
+  kphp_assert(request_interface);
+  if (!request_interface->is_parent_of(klass)) {
+    return;
+  }
+  if (compile_declaration_only) {
+    FunctionSignatureGenerator(W).set_overridden().set_const_this() << "class_instance<C$KphpJobWorkerSharedMemoryPiece> get_shared_memory_piece()" << SemicolonAndNL{};
+    FunctionSignatureGenerator(W).set_overridden() << "void set_shared_memory_piece(const class_instance<C$KphpJobWorkerSharedMemoryPiece> &instance)" << SemicolonAndNL{};
+    return;
+  }
+  const auto *field = G->get_job_worker_shared_memory_piece(klass->name);
+  W << NL;
+  FunctionSignatureGenerator(W).set_const_this().set_inline()
+    << "class_instance<C$KphpJobWorkerSharedMemoryPiece> " << klass->src_name << "::get_shared_memory_piece() " << BEGIN;
+  W << "return " << (field ? field->get_hash_name() : "{}") << SemicolonAndNL{};
+  W << END << NL << NL;
+
+  FunctionSignatureGenerator(W).set_inline() << "void " << klass->src_name
+                                                     << "::set_shared_memory_piece(const class_instance<C$KphpJobWorkerSharedMemoryPiece> &instance) " << BEGIN;
+  if (field) {
+    W << "const auto &casted = instance.cast_to<" << field->get_inferred_type()->class_type()->src_name << ">();" << NL;
```

Comment:
I suggest saving it by value, and using `std::move(casted)` bellow (so we use only move semantic)

Reply:
done